ta4j-core/src/main/java/org/ta4j/core/Indicator.java

```diff
@@ -53,21 +53,22 @@
 	Num numOf(Number number);
 
 	 /**
-	 * Returns all values from an {@link Indicator} as a Array of Doubles. The
+	 * Returns all values from an {@link Indicator} as an array of Doubles. The
 	 * returned doubles could have a minor loss of precise, if {@link Indicator}
 	 * was based on {@link Num Num}.
 	 *
 	 * @param ref the indicator
 	 * @param index the index
 	 * @param barCount the barCount
-	 * @return array of double within the barCount
+	 * @return array of Doubles within the barCount
 	 */
 	static Double[] toDouble(Indicator<Num> ref, int index, int barCount) {
 
 		Double[] all = new Double[barCount];
 
-		for (int i = Math.max(0, index - barCount + 1); i <= index; i++) {
-			Num number = ref.getValue(index);
+		int startIndex = Math.max(0, index - barCount + 1);
+		for (int i = 0; i < barCount; i++) {
+			Num number = ref.getValue(i + startIndex);
```

Comment:
Can you explain, please? What did you fix?

Reply:
@team172011 If I understand this method correctly, it should return an array of values ranged between `index - barCount + 1` and `index` but as you can see it's collecting the value at `index` multiple times and indexing `all` is also wrong. The added test shows what I expect from this method. Am I right?